src/wasm_standalone.cmake

```diff
@@ -9,12 +9,28 @@ set(CMAKE_SYSTEM_PROCESSOR wasm32)
 set(triple wasm32-unknown-unknown-wasm)
 
 set(WASM_SDKROOT ${CMAKE_CURRENT_LIST_DIR})
-set(CMAKE_C_COMPILER ${WASM_SDKROOT}/bin/clang)
-set(CMAKE_CXX_COMPILER ${WASM_SDKROOT}/bin/clang++)
-set(CMAKE_AR ${WASM_SDKROOT}/bin/llvm-ar)
-set(CMAKE_RANLIB ${WASM_SDKROOT}/bin/llvm-ranlib)
+
+if (CMAKE_HOST_WIN32)
+        set(EXE_SUFFIX ".exe")
+else()
+        set(EXE_SUFFIX "")
+endif()
+if ("${CMAKE_C_COMPILER}" STREQUAL "")
+  set(CMAKE_C_COMPILER ${WASM_SDKROOT}/bin/clang${EXE_SUFFIX})
+endif()
+if ("${CMAKE_CXX_COMPILER}" STREQUAL "")
+  set(CMAKE_CXX_COMPILER ${WASM_SDKROOT}/bin/clang++${EXE_SUFFIX})
+endif()
+if ("${CMAKE_AR}" STREQUAL "")
+  set(CMAKE_AR ${WASM_SDKROOT}/bin/llvm-ar${EXE_SUFFIX} CACHE FILEPATH "llvm ar")
```

Comment:
What are the last 3 args here doing and why are they only needed for AR and RANLIB?

Reply:
I'm not 100% sure why it's different; I cargo-culted this from emscripten (https://github.com/kripken/emscripten/blob/incoming/cmake/Modules/Platform/Emscripten.cmake#L82) The CACHE bit causes the path to go into CMakeCache.txt at configure time (and there's some interaction between that and when the toolchain cmake file is evaluated, which happens more than once). But I don't know why ar is special compared to the compiler.